src/link/http/iterators/async.ts

```diff
@@ -0,0 +1,13 @@
+export default function asyncIterator<T>(
+  source: AsyncIterableIterator<T>
+): AsyncIterableIterator<T> {
+  const iterator = source[Symbol.asyncIterator]();
+  return {
+    next(): Promise<IteratorResult<T, boolean>> {
+      return iterator.next();
+    },
+    [Symbol.asyncIterator](): AsyncIterator<T> {
+      return this;
+    },
+  } as AsyncIterableIterator<T>;
```

Comment:
```suggestion
    [Symbol.asyncIterator](): AsyncIterableIterator<T> {
      return this;
    },
  };
```
This seemed to work for me (no need for the final cast). What do you think?

Reply:
LGTM :) Updated